docs/se/guides/40_webclient.adoc

```diff
@@ -0,0 +1,420 @@
+
+[source,java]
+.Add the metric service to the WebClient:
+----
+WebClient webClient = WebClient.builder()
+                .baseUri("http://localhost:8080")
+                .addMediaSupport(JsonpSupport.create())
+                .addService(clientServiceMetric)           // <1>
+                .build();
+----
+<1> Register the metric service to the webclient.
+
+Simply use the `addService` method to add the metric to the WebClient on which the metrics will be measured.
+
+[source,java]
+.Print the metric count at the end of the main method:
+----
+System.out.println(metricName + ": " + counter.getCount());
+----
+
+To quickly check metrics are set up correctly, print the counter at the end of the main method. In this guide, the
+WebClient uses GET and PUT requests, so metrics can be applied on.
+
+==== Set up metrics with configuration files
+
+Using configuration file can reduce the code complexity and makes the metrics simpler to use. There is no
+need to modify the source code but only the configuration file to measure other values. The `application.yaml` file
+is the default configuration file for Helidon. It can be used to set up metrics settings.
+
+[source,yaml]
+.Example of metric configuration:
+----
+client:
+  services:
+    config:
+      metrics:
+        - type: METER
+          name-format: "client.meter.overall"
+        - type: TIMER
+          # meter per method
+          name-format: "client.meter.%1$s"
+        - methods: ["GET"]
+          type: COUNTER
+          errors: false
+          name-format: "client.counter.%1$s.success"
+          description: "Counter of successful GET requests"
+        - methods: ["PUT", "POST", "DELETE"]
+          type: COUNTER
+          success: false
+          name-format: "wc.counter.%1$s.error"
+          description: "Counter of failed PUT, POST and DELETE requests"
+        - methods: ["GET"]
+          type: GAUGE_IN_PROGRESS
+          name-format: "client.inprogress.%2$s"
+          description: "In progress requests to host"
+----
+
+The metrics are located under `client.services.config.metrics`. The metric setting can start either by its type or
+methods. The configuration file uses the same keywords as the programmatic way. `type` defines the kind of metric.
+
+[source,java]
+.Add the metric service to the WebClient:
+----
+Config config = Config.create();       // <1>
+
+WebClient webClient = WebClient.builder()
+                .baseUri("http://localhost:8080")
+                .config(config.get("client"))       // <2>
+                .addMediaSupport(JsonpSupport.create())
+                .build();
+----
+<1> Create a Helidon Config instance from default file `application.yaml`.
+<2> Configure the webclient with the `client` section from `application.yaml`.
+
+As shown, using configuration file strongly reduces the amount of code needed in the source code. If you want more
```

Comment:
WebClient

As demonstrated, using the configuration file reduces the amount of code needed in the source code. For more information about metrics, see the <<se/guides/05_metrics.adoc,Helidon Metrics Guide>>.